src/util/mod.rs

```diff
@@ -222,6 +234,88 @@ impl VersionTriple {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum VersionNumberError {
+    #[error("Failed to parse extra version version from {version:?}: {source}")]
+    ExtraVersionINvalid {
+        version: String,
+        source: std::num::ParseIntError,
+    },
+    #[error("Failed to parse version triple.")]
+    VersionTriplerError(#[from] VersionTripleError),
+}
+
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct VersionNumber {
```

Comment:
In the context that I haven't investigated the required formatting for Android version numbers, so I restricted it to iOS since this format might not quite work on Android. I can look into that a bit tho.